tools/cli/lib/api.c

```diff
@@ -671,6 +671,322 @@ TDNFCliRepoSyncCommand(
+       BAIL_ON_CLI_ERROR(dwError);
+       *ppCmdArgs=pCmdArgs;
+    } else {
+       pCmdArgs=*ppCmdArgs;
+    }
+
+    if(pCmdArgs->nCmdCount > 0)
+    {
+       if (pCmdArgs->pszQueryFormat_dep == false)
+           pCmdArgs->pszQueryFormat_dep = true;
+
+       dwError = TDNFQueryFormatDepCommand(pContext, pCmdArgs, data);
+       BAIL_ON_CLI_ERROR(dwError);
+    }
+
+cleanup:
+    pCmdArgs->pszQueryFormat_dep = false;
+    if((pCmdArgs && (tdwCount == (pdwCount+1))) || (dwError != 0))
+    {
+        ppCmdArgs=NULL;
+        TDNFFreeCmdArgs(pCmdArgs);
+    }
+    return dwError;
+
+error:
+    if (dwError == ERROR_TDNF_CLI_NOTHING_TO_DO ||
+        dwError == ERROR_TDNF_NO_DATA)
+    {
+        // Nothing to do should not return an error code
+        dwError = 0;
+    }
+
+    goto cleanup;
+}
+
+uint32_t tag_handler(PTDNF_CLI_CONTEXT pContext, const char *tag, char **pszRet, void *data, const int tdwCount, const int pdwCount, PTDNF_CMD_ARGS* ppCmdArgs)
+{
+    PTDNF_PKG_INFO pPkgInfo = (PTDNF_PKG_INFO)data;
+    char *pszVal = NULL;
+    uint32_t dwError = 0;
+    int nquery_arg;
+
+    if (strcmp(tag, "name") == 0) {
+        pszVal = pPkgInfo->pszName;
+    } else if (strcmp(tag, "arch") == 0) {
+        pszVal = pPkgInfo->pszArch;
+    } else if (strcmp(tag, "version") == 0) {
+        pszVal = pPkgInfo->pszVersion;
+    } else if (strcmp(tag, "reponame") == 0) {
+        pszVal = pPkgInfo->pszRepoName;
+    } else if (strcmp(tag, "release") == 0) {
+        pszVal = pPkgInfo->pszRelease;
+    } else if (strcmp(tag, "evr") == 0) {
+        pszVal = pPkgInfo->pszEVR;
+    } else if (strcmp(tag, "sourcename") == 0) {
+        pszVal = pPkgInfo->pszSourcePkg;
+    } else if (strcmp(tag, "size") == 0) {
+        pszVal = pPkgInfo->pszFormattedDownloadSize;
+    } else if (strcmp(tag, "downloadsize") == 0) {
+        pszVal = pPkgInfo->pszFormattedDownloadSize;
+    } else if (strcmp(tag, "installsize") == 0) {
+        pszVal = pPkgInfo->pszFormattedSize;
+    } else if (strcmp(tag, "sourcerpm") == 0) {
+        pszVal = pPkgInfo->pszSourcePkg;
+    } else if (strcmp(tag, "description") == 0) {
+        pszVal = pPkgInfo->pszDescription;
+    } else if (strcmp(tag, "summary") == 0) {
+        pszVal = pPkgInfo->pszSummary;
+    } else if (strcmp(tag, "license") == 0) {
+        pszVal = pPkgInfo->pszLicense;
+    } else if (strcmp(tag, "url") == 0) {
+        pszVal = pPkgInfo->pszURL;
+    } else if (strcmp(tag, "requires") == 0) {
+        nquery_arg = 8;
+        char *query_argv[] = {"", "", "", "", "", "repoquery", "--requires", pPkgInfo->pszName};
+        dwError = find_dependencies(pContext, nquery_arg, query_argv, data, tdwCount, pdwCount, ppCmdArgs);
+        BAIL_ON_CLI_ERROR(dwError);
+        pszVal = pPkgInfo->ppszDependencies[0];
+    } else if (strcmp(tag, "provides") == 0) {
+        nquery_arg = 8;
+        char *query_argv[] = {"", "", "", "", "", "repoquery", "--provides", pPkgInfo->pszName};
```

Comment:
Why do we need these empty args?

Anyway, this is not how we should do this.

Reply:
Removed the code for reparsing the arguments